src/Debugger/Daemon.php

```diff
@@ -0,0 +1,173 @@
+<?php
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Debugger;
+
+use Google\Cloud\Core\Exception\ConflictException;
+use Google\Cloud\Debugger\BreakpointStorage\BreakpointStorageInterface;
+use Google\Cloud\Debugger\BreakpointStorage\SysvBreakpointStorage;
+
+/**
+ * This class is responsible for registering itself as a Debuggee with the
+ * Stackdriver backend. It will fetch the list of breakpoints from the
+ * Stackdriver backend, validate and normalize them, and store them into the
+ * configured breakpoint storage.
+ *
+ * Example:
+ * ```
+ * use Google\Cloud\Debugger\Daemon;
+ *
+ * $daemon = new Daemon('/path/to/source/root');
+ * $daemon->run();
+ * ```
+ */
+class Daemon
+{
+    /**
+     * @var Debuggee
+     */
+    private $debuggee;
+
+    /**
+     * @var string The full path to the source root
+     */
+    private $sourceRoot;
+
+    /**
+     * @var BreakpointStorageInterface
+     */
+    private $storage;
+
+    /**
+     * Creates a new Daemon instance.
+     *
+     * @param string $sourceRoot The full path to the source root
+     * @param array $options [optional] {
+     *      Configuration options.
+     *
+     *      @type DebuggerClient $client A DebuggerClient to use. **Defaults
+     *            to** a new DebuggerClient.
+     *      @type array $extSourceContext The source code identifier. **Defaults
+     *            to** values autodetected from the environment.
+     *      @type string $uniquifier A string when uniquely identifies this
+     *            debuggee. **Defaults to** a value autodetected from the
+     *            environment.
+     *      @type BreakpointStorageInterface $storage The breakpoint storage
+     *            mechanism to use.
+     *      @type bool $debugOutput Whether or not to enable debug output.
+     * }
+     */
+    public function __construct($sourceRoot, array $options = [])
```

Comment:
Just a few notes, comparing this to https://github.com/chingor13/google-cloud-php/blob/ae4daca2fcd7c14d7ba88c180e71685bda91e50c/src/Core/Batch/BatchDaemon.php

- Do you see any value in checking early for the sysv extensions?
- Any value in setting up signal handlers?